erpnext/hr/doctype/upload_attendance/upload_attendance.py

```diff
@@ -119,6 +118,14 @@ def upload():
 	from frappe.utils.csvutils import read_csv_content_from_uploaded_file
 	from frappe.modules import scrub
 
+	if getattr(frappe, "uploaded_file", None):
+		fname = frappe.uploaded_file
+	else:
+		fname = frappe.form_dict.filename
+
+	if not fname.lower().endswith(".csv"):
```

Comment:
Add a simple check for `fname`, just to see if it's not empty, otherwise it'll fail at the `lower` call